tests/modules/get_all_users/app/test_get_all_users_controller.py

```diff
@@ -0,0 +1,26 @@
+from src.modules.get_all_users.app.get_all_users_controller import GetAllUsersController
+from src.modules.get_all_users.app.get_all_users_usecase import GetAllUsersUsecase
+from src.shared.infra.repositories.user_repository_mock import UserRepositoryMock
+
+
+class Test_GetAllUsersController:
+
+    def test_get_all_users_controller(self):
+        repo_mock = UserRepositoryMock()
+        get_all_users_usecase = GetAllUsersUsecase(repo_mock)
+        controller = GetAllUsersController(get_all_users_usecase)
+
+        response = controller()
+
+        assert response.status_code == 200
+
+    def test_controller_must_return_404_when_No_items_found(self):
+        repo_mock = UserRepositoryMock()
+        repo_mock.users = []
+        get_all_users_usecase = GetAllUsersUsecase(repo_mock)
+        controller = GetAllUsersController(get_all_users_usecase)
+
+        response = controller()
+
+        assert response.status_code == 404
```

Comment:
esse vai apagar por conta de manter lista [] qnd não tiver itens no mock, mas qnd for testar erro, coloca pelo menos teste da mensagem que levanta o erro (um status_code 404 pode indicar diversos erros)

Reply:
OK